qiskit/providers/ibmq/ibmqbackend.py

```diff
@@ -170,10 +254,45 @@ def run(
             api_job_share_level = ApiJobShareLevel.NONE
 
         validate_job_tags(job_tags, IBMQBackendValueError)
+
+        if isinstance(circuits, (QasmQobj, PulseQobj)):
+            warnings.warn("Passing a Qobj to Backend.run is deprecated and will "
+                          "be removed in a future release. Please pass in circuits "
+                          "or pulse schedules instead. The `qobj` parameter has been "
+                          "renamed to `circuits`.", DeprecationWarning,
+                          stacklevel=2)
+            qobj = circuits
+        else:
+            qobj_header = run_config.pop('qobj_header', None)
+            header = header or qobj_header
+            run_config_dict = self._get_run_config(
+                qobj_header=header,
+                shots=shots,
+                memory=memory,
+                qubit_lo_freq=qubit_lo_freq,
+                meas_lo_freq=meas_lo_freq,
+                schedule_los=schedule_los,
+                meas_level=meas_level,
+                meas_return=meas_return,
+                memory_slot_size=memory_slot_size,
+                rep_time=rep_time,
+                rep_delay=rep_delay,
+                init_qubits=init_qubits,
+                **run_config)
+            qobj = assemble(circuits, self, **run_config_dict)
+
         if validate_qobj:
             validate_qobj_against_schema(qobj)
         return self._submit_job(qobj, job_name, api_job_share_level, job_tags)
 
+    def _get_run_config(self, **kwargs: Any) -> Dict:
+        """Return the consolidated runtime configuration."""
+        run_config_dict = copy.copy(self.options.__dict__)
+        for key, val in kwargs.items():
+            if val:
+                run_config_dict[key] = val
+        return run_config_dict
```

Comment:
I'm wondering if we want to add something here to warn if somebody is setting something that isn't a supported option. I realize you could dump anything in the qobj run_config before, but only the things specified in the options object should be used now having extra fields is non-fatal, but just don't do anything right?